### DISCUSSION TITLE
Mistake : Declaration of Yajra\Pdo\Oci8::

### DISCUSSION THREAD

**Discussion by romarioagros** [yajra/laravel-oci8#676] [Q&A]
When I try to query like this:

<?php

namespace App\Http\Controllers\Voice;
use Auth;
use Illuminate\Http\Request;
use App\Models\Contact;
use App\Http\Requests\ContactRequest;
use Illuminate\Support\Facades\DB;
use App\Http\Controllers\Controller;
use App\Exports\ContactExport;

use Maatwebsite\Excel\Facades\Excel;

class TestController extends Controller{
    
    public function test(){

  
      $new = DB::connection('oracle')->select("
      select   * from base  where ROWNUM <= 5");

      print_r($new) ;
  
  

}
}

PHP 8 , laravel 8 , I receive this mistake :
Declaration of Yajra\Pdo\Oci8::query($statement, $fetchMode = null, $modeArg = null, $ctorArgs = null) must be compatible with PDO::query(string $query, ?int $fetchMode = null, mixed ...$fetchModeArgs)

**Comment by yajra** [comment]
For PHP8, please make sure you use pdo-via-oci8 v3.x https://github.com/yajra/pdo-via-oci8

**Reply by romarioagros** [reply]
thank you 
I install it 
but how i can use it ?
What I can fill instead of "providers' =>
 [Yajra\Oci8\Oci8ServiceProvider::class,] ?

**Reply by yajra** [reply]
Maybe try reinstalling via `composer require yajra/laravel-pdo-via-oci8:3.*`?

**Reply by romarioagros** [reply]
Thank you very much!
It is working now!
